aten/src/ATen/native/mkldnn/MKLDNNConversions.cpp

```diff
@@ -26,6 +26,7 @@ Tensor dense_to_mkldnn(const Tensor& cpu_tensor) {
              "dense_to_mkldnn expects float tensor input");
   AT_ASSERTM(cpu_tensor.dim() <= 5,
              "Can't convert cpu tensor with the number of dimensions > 5");
+  TORCH_CHECK(at::globalContext().userEnabledMkldnn(), "mkldnn backend was disabled by users");
```

Comment:
I wouldn't actually put this test in here. MKLDNN "disabled" just means we won't select it when we're trying to make a kernel choice. If someone explicitly calls an mkldnn function, we should still play ball and let them use it.

Reply:
OK, that's something we struggle to do or not to do. Just wait this opinion 😆.

Reply:
@ezyang The reason why we checked the flag here was that we wanted to make the API semantics consistent, i.e. MKL-DNN backend disabled meaning MKL-DNN backend won't be dispatched either explicitly or implicitly.

Reply:
Yeah, that's not what the semantics of the cudnn flag are.

```
>>> torch.cudnn_convolution(torch.zeros(20,33,50,100,device='cuda'), torch.zeros(16,33,3,3,device='cuda'), None, (0,0), (2,2), (1,1), 1, False, False).size()
torch.Size([20, 16, 24, 49])
>>> import torch.backends.cudnn
>>> torch.backends.cudnn.disabled = True
>>> torch.cudnn_convolution(torch.zeros(20,33,50,100,device='cuda'), torch.zeros(16,33,3,3,device='cuda'), None, (0,0), (2,2), (1,1), 1, False, False).size()
torch.Size([20, 16, 24, 49])
```
 
One important thing to note is that the intended use of the runtime disable flag is different from the use of the compile time disable flag. The compile time disable flag is to prevent us from building MKLDNN/cuDNN at all; that means that both the generic convolution and the specific cudnn_convolution functions cannot work, since you don't have the code at all.  But with a runtime flag, we do have the code available, so there is no reason to make cudnn_convolution stop working when you have the code. The main use case for the flag is to just stop pathways using the generic convolution (i.e., most of them) from using mkldnn. Hope that makes sense.

Reply:
@ezyang Thanks for the explanation. We will have it aligned with `torch.backends.cudnn.disabled`.